sdks/java/harness/src/main/java/org/apache/beam/fn/harness/BoundedSourceRunner.java

```diff
@@ -124,10 +128,17 @@ public void start() throws Exception {
     try {
       // The representation here is defined as the java serialized representation of the
       // bounded source object in a ByteString wrapper.
-      byte[] bytes = definition.getPayload().toByteArray();
-      @SuppressWarnings("unchecked")
-      InputT boundedSource =
-          (InputT) SerializableUtils.deserializeFromByteArray(bytes, definition.toString());
+      InputT boundedSource;
+      if (definition.getUrn().equals(ProcessBundleHandler.JAVA_SOURCE_URN)) {
+        byte[] bytes = definition.getPayload().toByteArray();
+        @SuppressWarnings("unchecked")
+        InputT boundedSource0 =
+            (InputT) SerializableUtils.deserializeFromByteArray(bytes, definition.toString());
+        boundedSource = boundedSource0;
+      } else {
```

Comment:
<!--new_thread; commit:5300be61ccac7a006ec7f2b8f3824cddb4dc75e8; resolved:0-->
It kind of looks like this branch is expected to have READ_URN. Is that so? Does that not work? I'm just curious, and would love to have the bare `else` case be an error, with all understood cases being guarded by a URN.

Reply:
That is correct (given the factory registration above), but made more explicit.